pkg/database/mariadb.go

```diff
@@ -10,75 +10,116 @@ import (
 	"strings"
 
 	_ "github.com/go-sql-driver/mysql"
+	"github.com/jmoiron/sqlx"
 	"github.com/kubearchive/kubearchive/pkg/models"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
 func init() {
 	RegisteredDatabases["mariadb"] = NewMariaDBDatabase
+	RegisteredDBCreators["postgresql"] = NewMariaDBCreator
```

Comment:
I think you meant to change the key to `mariadb`
```suggestion
	RegisteredDBCreators["mariadb"] = NewMariaDBCreator
```